iotilecore/iotile/core/hw/auth/auth_chain.py

```diff
@@ -45,136 +44,33 @@ def _load_installed_providers(self):
         for name, entry in reg.load_extensions('iotile.auth_provider'):
             self._auth_factories[name] = entry
 
-    def encrypt_report(self, device_id, root, data, **kwargs):
-        """Encrypt a buffer of report data on behalf of a device.
-
-        Args:
-            device_id (int): The id of the device that we should encrypt for
-            root (int): The root key type that should be used to generate the report
-            data (bytearray): The data that we should encrypt.
-            **kwargs: There are additional specific keyword args that are required
-                depending on the root key used.  Typically, you must specify
-                - report_id (int): The report id
-                - sent_timestamp (int): The sent timestamp of the report
-
-                These two bits of information are used to construct the per report
-                signing and encryption key from the specific root key type.
-
-        Returns:
-            dict: The encrypted data and any associated metadata about the data.
-                The data itself must always be a bytearray stored under the 'data'
-                key, however additional keys may be present depending on the encryption method
-                used.
-
-        Raises:
-            NotFoundError: If the auth provider is not able to encrypt the data.
-        """
-
+    def get_root_key(self, key_type, device_id):
```

Comment:
I got further down in the review and noticed that there are very similarly named functions that have docstrings in other classes (like base classes or other key modes I guess). 

I think that we still want to define the docstrings in these methods, because I can't think of a way to see the method we overwrote (and it's not an `abc`).

Reply:
Thank you for suggestions, you have already mentioned docstrings in your previous comments to other my MRs, I will strive to fix this style issues before passing it to review